### PR TITLE
Completed Unit Test: Loads Comments on First Render

### PR DESCRIPTION
**Pull request by Eshah7**
Tests if the comments render when the page loads!

## Trello ticket link
N/A

## Implementation description
* Created a ForumTest page to bypass the restrictions imposed by Firebase's API
* Created a test to see if the comments render upon the creation of the page for a forum (id = 1 for this purpose) 

## Steps to test
1. npm run test in Terminal

## What should reviewers focus on?
* The tests pass: 
image

### CODE REVIEW

File: src/tests/getForumCommentsByForumID.spec.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+describe('<ForumTest />', () => {
+
+  it('loads comments on first render', async () => {
+    global.fetch = jest.fn(() => {
+      Promise.resolve();
+    })
+
+    const loadGetForumCommentsByForumID = jest.fn().mockName('loadGetForumCommentsByForumID');
+    render(<Forum loadGetForumCommentsByForumID={loadGetForumCommentsByForumID}/>);
+    expect(loadGetForumCommentsByForumID).toHaveBeenCalled();
```
Comment (safchow) [inline src/tests/getForumCommentsByForumID.spec.js]: add a few more tests -> a lot of them are reusable and require one/two line changes, look at the tests in https://github.com/Safewaan/HonkedIn/pull/32 for reference

**Review by safchow — CHANGES_REQUESTED** [review]

File: src/tests/getForumCommentsByForumID.spec.js (hunk 1 above)
Comment (Eshah7) [reply src/tests/getForumCommentsByForumID.spec.js]: Re-requested review to merge the one working unit-test for now - thanks!

File: package.json [hunk 2]
```diff
@@ -12,9 +12,10 @@
     "@testing-library/user-event": "^7.2.1",
     "babel-jest": "^27.0.0",
     "bootstrap": "^4.6.2",
+    "fetch-mock": "^9.11.0",
```
Comment (safchow) [inline package.json:15]: note for next time: please double check any added dependencies if they're being used

this dependency is not being used so it should be removed from `package.json` and `package-lock.json` but I'll clean this up once all PRs are merged

File: src/componentsTest/ForumTest.js [hunk 3]
```diff
@@ -0,0 +1,261 @@
+import React, { useRef, useState, useEffect } from "react"
```
Comment (safchow) [inline src/componentsTest/ForumTest.js]: this file should retain its original name: `Forum.js`

File: src/tests/getForumCommentsByForumID.spec.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
```
Comment (safchow) [inline src/tests/getForumCommentsByForumID.spec.js]: This file should be called `Forum.test.js`

File: src/componentsTest/ForumTest.js [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+                        <CardContent>
+                            {/*<Link to={`/forum/${event.id}`} target="_blank">*/}
+                            <Typography variant="h5" component="div">
+                                {event.forumTitle}<br />
+                            </Typography>
+                            {/*</Link>*/}
+                            <Typography sx={{ mb: 1.5 }} color="text.secondary">
+                                Posted on {new Date(event.dateTime).toLocaleDateString()}<br />
+                            </Typography>
+                            <Typography sx={{ mb: 1.5 }} color="text.secondary">
```
Comment (safchow) [inline src/componentsTest/ForumTest.js]: remove the `color=text.secondary` styling from all components as it throws a warning during test runs

**Review by safchow — APPROVED** [review]

**Pull request merged**